src/Equinox.Core/Cache.fs

```diff
@@ -1,61 +1,27 @@
 ﻿namespace Equinox.Core
 
-open System
-open System.Threading.Tasks
-
 type [<NoEquality; NoComparison; Struct>] CacheItemOptions =
-    | AbsoluteExpiration of ae: DateTimeOffset
-    | RelativeExpiration of re: TimeSpan
-
-[<AllowNullLiteral>]
-type CacheEntry<'state>(initialToken: StreamToken, initialState: 'state) =
-    let mutable currentToken = initialToken
-    let mutable currentState = initialState
-    member x.UpdateIfNewer(supersedes: struct (StreamToken * StreamToken) -> bool, other: CacheEntry<'state>) =
-        lock x <| fun () ->
-            let struct (otherToken, otherState) = other.Value
-            if supersedes (currentToken, otherToken) then
-                currentToken <- otherToken
-                currentState <- otherState
-
-    member x.Value: struct (StreamToken * 'state) =
-        lock x <| fun () ->
-            currentToken, currentState
+    | AbsoluteExpiration of ae: System.DateTimeOffset
+    | RelativeExpiration of re: System.TimeSpan
 
 type ICache =
-    abstract member UpdateIfNewer: key: string * options: CacheItemOptions * (struct (StreamToken * StreamToken) -> bool) * entry: CacheEntry<'state> -> Task<unit>
-    abstract member TryGet: key: string -> Task<struct (StreamToken * 'state) voption>
+    abstract member TryGet:
+                          key: string
+                           -> System.Threading.Tasks.Task<struct (StreamToken * 'state) voption>
+    abstract member UpdateIfNewer:
+                          key: string
+                          * compareTokens:(struct (StreamToken * StreamToken) -> int64)
```

Comment:
🤔 could this be `StreamToken -> StreamToken -> StreamToken`?

Reply:
YES, thank you - it's one unnecessarily ugly duck as it is
But it's a `struct` (and prob should stay that), so the caller would need to do a `ReferenceEquals` on `.value` to be able to identify when a superseded value was ignored.
I guess `-> StreamToken voption` can convey that more cleanly

Reply:
went with isStale  - it's in the negative, but is meaningful to someone scanning (versus `isValid`), thanks for prompting!